canonicalwebteam/blog/views.py

```diff
@@ -396,6 +396,21 @@ def _get_article_context(
             if set(related_tag_ids) <= set(related_article["tags"]):
                 related_articles.append(related_article)
 
+            # Iterate over the tags for the current article and counting
+            # matches against related articles
+            for tag in tags:
+                related_article["compatibility"] = related_article[
+                    "tags"
+                ].count(tag["id"])
```

Comment:
Not sure I understand how this works.

- `tags` is the tags of the current article.
- You iterate over `tags` to see how many of them match the tags of the `related_article`
- The 3 `related_articles` with the most matching tags are displayed on the page 

I am not sure if this achieves it. Wouldn't `related_article["compatibility"]` always save the last `count(tag["id"])`? Perhaps `+=` would fix it.

I suggest this: https://www.w3schools.com/python/ref_set_intersection.asp
Maybe it would work better.

Reply:
@albertkol good points, I was definitely over looping here. I hope this update makes it much more clear and thanks for the hit on the intersection function.